lib/python/rose/suite_clean.py

```diff
@@ -102,11 +102,37 @@ def _clean(self, suite_name, only_items=None):
                     if item:
                         loc_rel = os.path.join(suite_dir_rel, item)
                     locs.append(os.path.join(item_root, loc_rel))
+                    roots.add(item_root)
             if self.host_selector.is_local_host(auth):
+                # Clean relevant items
                 for loc in locs:
                     loc = os.path.abspath(env_var_process(loc))
                     for name in sorted(glob(loc)):
                         engine.fs_util.delete(name)
+                # Clean empty directories
+                # Change directory to root level to avoid cleaning them as well
+                # For cylc suites, e.g. it can clean up to an empty "cylc-run/"
+                # directory.
+                for root in sorted(roots):
+                    cwd = os.getcwd()
+                    if root:
+                        try:
+                            os.chdir(env_var_process(root))
+                        except OSError:
+                            continue
+                    # Reverse sort to ensure that e.g. "share/cycle/" is
+                    # cleaned before "share/"
+                    for name in sorted(self.CLEANABLE_PATHS, reverse=True):
```

Comment:
This doesn't seem particularly friendly/nice/pythonic as a general practice as it makes alterations to CLEANABLE_PATHS problematic at this point in the code. It'd be much better if you altered CLEANABLE_PATHS to be in the desired order of precedence with a note in a comment regarding the importance of maintaining said order.